test/e2e/functional/tests/e2e.go

```diff
@@ -707,39 +735,17 @@ var _ = Describe("metallb", func() {
 
 			Expect(isAddresspoolValidationWebhookRunning).To(BeTrue(), "AddressPool webhook is not running")
 		})
+		AfterEach(func() {
+			_ = testclient.Client.Delete(context.Background(), firstAddressPool)
+			_ = testclient.Client.Delete(context.Background(), secondAdressPool)
+		})
 		It("Should recognize overlapping addresses in two AddressPools", func() {
 			By("Creating first AddressPool resource")
-			autoAssign := false
```

Comment:
If these values are local to this test, and not reused by other tests, I think it makes sense to keep them here and not globally.